docs/_guide/properties.md

```diff
@@ -28,158 +78,232 @@ static get properties() {
 
-## Configure corresponding attributes
+### Initialize property values when using TypeScript decorators
 
-Configure a property's corresponding attributes and attribute behavior with the `type`, `attribute` and `reflect` options.
+TypeScript users can initialize property values when they are declared with the `@property` decorator:
 
-### Configure a property type
+```ts
+@property({ type : String }) prop1 = 'Hello World';
+```
 
-While element properties can be of any type, attributes are always strings. A property must be serialized and deserialized to and from its corresponding observed attribute. This is specified with the `type` option in a property declaration. For example:
+**Example: Initialize property values when using TypeScript decorators** 
 
 ```js
-myProp { type: Boolean }
+{% include projects/properties/inittypescript/my-element.ts %}
+```
+
+{% include project.html folder="properties/inittypescript" openFile="my-element.ts" %}
+
+### Initialize property values from attributes in markup 
+
+You can also initialize property values from observed attributes in markup:
+
+_index.html_ 
+
+```html
+<my-element 
+  mystring="hello world"
+  mynumber="5"
+  mybool
+  myobj='{"stuff":"hi"}'
+  myarray='[1,2,3,4]'></my-element>
 ```
 
-By default, LitElement uses the `String` constructor to serialize and deserialize properties and attributes. To make sure a non-string property is handled correctly, configure the property's `type` option.
+{% include project.html folder="properties/initmarkup" openFile="index.html" %}
+
+See [observed attributes](#observed-attributes) and [converting between properties and attributes](#conversion) for more information on setting up initialization from attributes.
 
-`type` can be: 
+## Configure attributes {#attributes}
 
-* A function that performs both serialization and deserialization:
+### Convert between properties and attributes {#conversion}
 
-  ```js
-  propName: { type: someFunction }
-  ```
+While element properties can be of any type, attributes are always strings. This impacts the [observed attributes](#observed-attributes) and [reflected attributes](#reflected-attributes) of non-string properties:
 
-* An object with two function properties, `fromAttribute` and `toAttribute`. `fromAttribute` performs deserialization, and `toAttribute` performs serialization:
+  * To **observe** an attribute (set a property from an attribute), the attribute value must be converted from a string to match the property type.
 
-  ```js
-  propName: { type: {
-    toAttribute: serializerFunction,
-    fromAttribute: deserializerFunction
-  }}
-  ```
+  * To **reflect** an attribute (set an attribute from a property), the property value must be converted to a string.
 
-By default, `type`, `fromAttribute` and `toAttribute` are the `String` constructor. 
+LitElement defines a **converter** for every declared property. A converter is an object that contains two functions:
```

Comment:
Maybe "LitElement _accepts_ a **converter** for every declared property"?

Also, a `converter` can be an object with two functions, or just a function (with `fromAttribute` semantics).